### PR TITLE
MVP. Full basic functionality.

### PR DESCRIPTION
**Pull request by adamryman**
- [X] OAuth2 flow with github to be able to make web hooks on a users repos
	- [X] [github.com/golang/oauth2](https://github.com/golang/oauth2)
	- [X] Any OAuth token
	- [X] Associate slack user with OAuth2 token
- [X] Github Webhooks
	- [X] [github.com/google/go-github](https://github.com/google/go-github)
	- [ ] Store slack webhook
	- [X] Add webhook to repo
		- [X] application/json
		- [X] Pull request
- [ ] Bot
	- [X] Connect to slack
		- [X] Do random stuff
		- [X] Move code out of `main.go` to `lgtm/bot`
	- [X] Flow that does oauth
	- [X] Flow that adds webhook
		- [X] `@lgtm Watch repo adamryman/kit`
	- [ ] Flow that removes webhook?
		- [ ] `@lgtm stop looking at adamryman/kit`
	- [X] Parse webhook response, post in #pr-party
		- [X] Look for `{ "action": "opened"}`
			- [X] Post in #pr-party
			- [ ] Store timestamp of message to react when closed
		- [X] Look for `{ "action": "closed"}`
			- [ ] React to old post in #pr-party

### CODE REVIEW

**Review by lelandbatey — APPROVED** [review]
## LOOKS GOOD TO ME :+1:

**Comment by tjhorner** [comment]
Wonderful PR would merge 11/10

**Review by tjhorner — APPROVED** [review]
astounding.

File: lgtm.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+}
+
+func handleBotEvents(authurl string, lgtm *bot.LGTM, database *sql.Database) {
+	ctx := context.Background()
+	for event := range lgtm.IncomingEvents {
+		switch ev := event.(type) {
+		case bot.WatchRepoEvent:
+			lgtm.PostMessage(fmt.Sprintf(
```
Comment (tylermenezes) [inline lgtm.go:161]: If you add an error message for when it's already watching the repo, then there's no case where it won't immediately send another message, and you could probably remove this.

**Review by tylermenezes — APPROVED** [review]
Never thought this day would come!

**Pull request merged**